src/mbgl/tile/geometry_tile_worker.cpp

```diff
@@ -323,9 +323,8 @@ void GeometryTileWorker::redoLayout() {
                 if (!filter(feature->getType(), feature->getID(), [&] (const auto& key) { return feature->getValue(key); }))
                     continue;
 
-                GeometryCollection geometries = feature->getGeometries();
```

Comment:
Please revert this; calling `getGeometries()` only once per feature was an important performance optimization. (The result is constructed on the fly and not cached by `VectorTileFeature`.)